### PR TITLE
Add documentation and improve validation for swaps quotes and some of its properties

### PR DESCRIPTION
**Pull request by danjm**
This PR does the following:

- improves the validation of `quote.trade.value`, `quote.sourceAmount` and `quote.destinationAmount` as returned by the metaswap-api `/trades` endpoint
- adds documentation for the objects returned by the the `/trades` endpoint
- corrects the type of the default values of `destinationAmount` and `sourceAmount` in quotesToRenderableData
- corrects the base parameter used when creating a BigNumber from `quote.trade.value` in view-quote.js (note that this previously did not cause any error/bug because if the first parameter to BigNumber is `0x` prefixed, then it will treat it as a hex, even if the second parameter is not 16)

### CODE REVIEW

File: ui/app/pages/swaps/swaps.util.js [hunk 1]
```diff
@@ -35,13 +35,18 @@ const getBaseApi = function (type) {
...
 const truthyString = (string) => Boolean(string?.length)
+const stringIsValidDecimalNumber = (string) => !isNaN(string) && string.match(/^[.0-9]+$/u) && !isNaN(parseFloat(string))
 const truthyDigitString = (string) => truthyString(string) && Boolean(string.match(/^\d+$/u))
 
 const QUOTE_VALIDATORS = [
   {
     property: 'trade',
     type: 'object',
-    validator: (trade) => trade && validHex(trade.data) && isValidAddress(trade.to) && isValidAddress(trade.from) && truthyString(trade.value),
+    validator: (trade) => trade &&
```
Comment (darkwing) [inline ui/app/pages/swaps/swaps.util.js:45]: Just for clarity, do we want to add this as a function like `stringIsValidDecimalNumber` is on line 38?  Maybe `const tradeIsValid = ....`.  Maybe just for clarity.

File: ui/app/pages/swaps/swaps.util.js [hunk 2]
```diff
@@ -35,13 +35,18 @@ const getBaseApi = function (type) {
 
 const validHex = (string) => Boolean(string?.match(/^0x[a-f0-9]+$/u))
 const truthyString = (string) => Boolean(string?.length)
+const stringIsValidDecimalNumber = (string) => !isNaN(string) && string.match(/^[.0-9]+$/u) && !isNaN(parseFloat(string))
```
Comment (darkwing) [inline ui/app/pages/swaps/swaps.util.js:38]: `string.match(/^[.0-9]+$/u)` allows more than one "."; also, do we need both `!isNan` checks?  Doesn't one identify a problem?

Also, I think we should try to make these functions that we can write tests for.

**Review by darkwing — CHANGES_REQUESTED** [review]
Have an idea for improvement, also a question about testing.

**Comment by danjm** [comment]
Closing for now. We should revisit this when we have time. @dan437 may wish to pick up some parts of this.

**Pull request closed without merging**